action.php

```diff
@@ -691,8 +691,18 @@ protected function page_depend_replacements($raw, $id) {
         $replace['@PAGEURL@'] = wl($id, $params, true, "&");
         $replace['@QRCODE@']  = $qr_code;
 
+        $content = '';
+
+        // let other plugins define their own replacements
+        $evdata = ['replace' => &$replace, 'content' => &$content];
+        $event = new Doku_Event('PLUGIN_DW2PDF_REPLACE', $evdata);
+        $event->advise_before();
+
         $content = str_replace(array_keys($replace), array_values($replace), $raw);
```

Comment:
I would wrap this line in an `if($event->advise_before())` so plugins can prevent the default replacement completely. `$content` would need to be initialized with `$raw` then.

Reply:
done in d824d23dffa98ae114a4bd74f4830b7aef707eac